### PR TITLE
TDP-4198 Use streams for folder

### PR DESCRIPTION
**Pull request by fhuaulme**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4198

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [x] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/603/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/603/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/608/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/608/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/610/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/610/)

File: dataprep-backend-service/src/main/java/org/talend/dataprep/folder/store/FolderRepository.java [hunk 1]
```diff
@@ -164,6 +165,15 @@
     Folder getFolderById(String folderId);
 
     /**
+     * Count the number of objects of <code>type</code> in given <code>folder</code>.
+     *
+     * @param folder the folder id.
+     * @param type the content type.
+     * @return The number of <code>type</code> objects in given <code>folder</code>. Returns 0 if folder does not exist.
```
Comment (fhuaulme) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/folder/store/FolderRepository.java]: This is for consistency with other folder repository methods (that do not throw exception when folder does not exist).

Comment (fhuaulme) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/folder/store/FolderRepository.java]: :-1:

**Pull request merged**